core/src/main/java/nl/nn/adapterframework/receivers/ReceiverBase.java

```diff
@@ -894,15 +894,26 @@ public String processRequest(IListener origin, String correlationId, String mess
 		Date tsReceived = null;
 		Date tsSent = null;
 		if (context!=null) {
-			//TODO ClassCasting Exceptions occur when using PipeLineSessionBase.setListenerParameters
-			tsReceived = (Date)context.get(IPipeLineSession.tsReceivedKey);
-			tsSent = (Date)context.get(IPipeLineSession.tsSentKey);
+			//ClassCasting Exceptions occur when using PipeLineSessionBase.setListenerParameters, hence these silly instanceof's
```

Comment:
Kunnen we dat niet op een nette manier fixen?

Reply:
Ik heb hier issue #770 voor aangemaakt